src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -389,6 +423,7 @@ public boolean setHeaders(HttpURLConnection conn, HeaderParser p, String raw) {
         if (algorithm == null || algorithm.isEmpty()) {
             algorithm = "MD5";  // The default, accoriding to rfc2069
         }
+        algorithm = algorithm.toUpperCase();
```

Comment:
Please use `toUpperCase(Locale.ROOT)` or `toUpperCase(Locale.ENGLISH)`.

Reply:
> Should we have a test case for "SHA-512-256" too?

If there's hardcoded example in the RFC, I will include it